autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingActivityRepository.java

```diff
@@ -18,32 +19,54 @@
 
 @EntityType(entityClass = ScalingActivity.class)
 @Transactional(Transactional.TxType.REQUIRED)
-public interface ScalingActivityRepository extends CrudRepository<ScalingActivity, Long> {
+public interface ScalingActivityRepository extends PagingAndSortingRepository<ScalingActivity, Long> {
 
-    @Query("SELECT st FROM ScalingActivity st WHERE st.activityCrn = :activityCrn")
-    Optional<ScalingActivity> findByActivityCrn(@Param("activityCrn") String activityCrn);
+    @Query("SELECT st FROM ScalingActivity st WHERE st.activityCrn = :activityCrn AND st.cluster.clusterPertain.tenant = :tenant")
+    Optional<ScalingActivity> findByActivityCrnAndTenant(@Param("activityCrn") String activityCrn, @Param("tenant") String tenant);
 
-    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId")
-    List<ScalingActivity> findAllByCluster(@Param("clusterId") Long clusterId);
+    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.stackName = :clusterName")
+    List<ScalingActivity> findAllByCluster(@Param("clusterName") String clusterName);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime < :startTimeBefore ORDER BY st.startTime DESC")
     List<ScalingActivity> findAllByClusterWithStartTimeBefore(@Param("clusterId") Long clusterId, @Param("startTimeBefore") Date startTimeBefore);
 
-    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime >= :startTimeAfter ORDER BY st.startTime DESC")
-    List<ScalingActivity> findAllByClusterWithStartTimeAfter(@Param("clusterId") Long clusterId, @Param("startTimeAfter") Date startTimeAfter);
+    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.stackName = :clusterName" +
```

Comment:
I'm not very sure what query gets executed here - specifically st.cluster.stackName - does this end up being translated into a join with the cluster table on the cluster_id field ?

Reply:
We have the cluster in the scalingActivity entity and in that table we have the stackName and stackCrn values which I am using for joing the scaling activity table as we have the clusterName and clusterCrn values as input.